IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetry.m

```diff
@@ -135,4 +210,125 @@ - (MSIDLastRequestTelemetrySerializedItem *)createSerializedItem
     return [[MSIDLastRequestTelemetrySerializedItem alloc] initWithSchemaVersion:[NSNumber numberWithInteger:self.schemaVersion] defaultFields:defaultFields errorInfo:self.errorsInfo platformFields:nil];
 }
 
+#pragma mark - Update object
+
+- (void)addErrorInfo:(MSIDRequestTelemetryErrorInfo *)errorInfo
+{
+    dispatch_barrier_async(_synchronizationQueue, ^{
+        if(errorInfo)
+        {
+            _errorsInfo = [_errorsInfo count] ? _errorsInfo : [NSMutableArray new];
+           [_errorsInfo addObject:errorInfo];
+        }
+        
+        [self saveTelemetryToDisk];
+    });
+}
+
+- (void)resetTelemetry
+{
+    dispatch_barrier_async(_synchronizationQueue, ^{
+        _errorsInfo = nil;
+        _silentSuccessfulCount = 0;
+        [self saveTelemetryToDisk];
+    });
+}
+
+#pragma mark - Private: Save To Disk
+
+- (void)saveTelemetryToDisk
+{
+    NSString *saveLocation = [self filePathToSavedTelemetry];
+    if (saveLocation)
+    {
+        NSData *dataToArchive = [NSKeyedArchiver msidArchivedDataWithRootObject:self requiringSecureCoding:YES error:nil];
+        
+        [dataToArchive writeToFile:saveLocation atomically:YES];
+    }
+}
+
+- (instancetype)initFromDecodedObjectWithSchemaVersion:(NSInteger)schemaVersion silentSuccessfulCount:(NSInteger)silentSuccessfulCount errorsInfo:(NSMutableArray<MSIDRequestTelemetryErrorInfo *>*) errorsInfo
+{
+    self = [super init];
+    if (self)
+    {
+        if (schemaVersion == currentSchemaVersion)
+        {
+            _schemaVersion = schemaVersion;
+            _silentSuccessfulCount = silentSuccessfulCount;
+            _errorsInfo = errorsInfo;
+            _synchronizationQueue = [self initializeDispatchQueue];
+        }
+        else
+        {
+            self = [self initInternal];
+        }
+        
+    }
+    return self;
+}
+
+- (NSString *)filePathToSavedTelemetry
+{
+    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
```

Comment:
let's maybe save it to temp directory instead and let's also make sure it is not backed up. Please follow https://developer.apple.com/icloud/documentation/data-storage/index.html


```
To ensure that backups are as efficient as possible, store your app’s data according to the following guidelines:
Only documents and other data that is user-generated, or that cannot otherwise be recreated by your application, should be stored in the <Application_Home>/Documents directory and will be automatically backed up by iCloud.
Data that can be downloaded again or regenerated should be stored in the <Application_Home>/Library/Caches directory. Examples of files you should put in the Caches directory include database cache files and downloadable content, such as that used by magazine, newspaper, and map applications.
Data that is used only temporarily should be stored in the <Application_Home>/tmp directory. Although these files are not backed up to iCloud, remember to delete those files when you are done with them so that they do not continue to consume space on the user’s device.
Use the "do not back up" attribute for specifying files that should remain on device, even in low storage situations. Use this attribute with data that can be recreated but needs to persist even in low storage situations for proper functioning of your app or because customers expect it to be available during offline use. This attribute works on marked files regardless of what directory they are in, including the Documents directory. These files will not be purged and will not be included in the user's iCloud or iTunes backup. Because these files do use on-device storage space, your app is responsible for monitoring and purging these files periodically.
```

Reply:
I changed the file directory to the temp directory. I initially also implemented setting the excludeBackup flag to true, but I took it out because according to the below documentation from the link you gave, it would be redundant.

`Data that is used only temporarily should be stored in the <Application_Home>/tmp directory. Although these files are not backed up to iCloud, remember to delete those files when you are done with them so that they do not continue to consume space on the user’s device.`